phantomgames/Home/HomeScreenRepository.swift

```diff
@@ -0,0 +1,32 @@
+//
+//  HomeScreenRepository.swift
+//  phantomgames
+//
+//  Created by Aphiwe Shozi on 2024/05/13.
+//
+
+import Foundation
+typealias HomeScreenResult = (Result<[Game], APIError>) -> Void
+
+protocol HomeScreenRepositoryType: AnyObject {
+    func fetchAPIImage(completion: @escaping(HomeScreenResult))
+    func fetchHomeResults(completion: @escaping (HomeScreenResult))
+    
+}
+
+class HomeScreenRepository: HomeScreenRepositoryType {
+    func fetchAPIImage(completion: @escaping (HomeScreenResult)) {
+        URLSession.shared.fetchingAPIImages(URL: Constants.Endpoints.search) {
+            games in completion(.success(games))
+        }
+    }
+    
+    func fetchHomeResults(completion: @escaping (HomeScreenResult)) {
+        URLSession.shared.request(endpoint: Constants.Endpoints.search, method: .GET, completion: completion)
+    }
+    
+}
+
+
```

Comment:
a whole bunch of new lines???
Please FIX the swiftlint warnings. I will be declining future PR's. I can't keep commenting this on every PR. Please pss the message to the team as well